File: back-end/spring/spring-api-rest/src/main/java/com/scaffolding/spring/model/EntityModel.java
package com.scaffolding.spring.model;

import lombok.Data;

import java.util.Date;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
@Entity
public class EntityModel {

	private @Id @GeneratedValue Long entityId;
	@NotNull
	@NotEmpty
	private String entityTitle;
	private String entityDescription;
	private Date createDate;
	private Date updateDate;
	private Date deleteDate;

	public EntityModel() {

	}

	public EntityModel(String entityTitle, String entityDescription) {
		this.entityTitle = entityTitle;
		this.entityDescription = entityDescription;
		this.createDate = new Date();
	}

}

File: front-end/web/ionic/src/app/models/entity.model.ts

export interface Entity {
    entityId?: number;
    entityCode: string;
    entityTitle: string;
    entityDescription?: string;
    entityTypeId?: number;
    createDate?: String;
    createUser?: number;
    updateDate?: String;
    updateUser?: number;
}

export const convertToEntity = (data: any): Entity => {

    const createDate = new Date(data.createDate);
    const updateDate = new Date(data.updateDate);

    const entity: Entity = {
      ...data,
      createDate: createDate.toISOString(),
      updateDate: updateDate.toISOString(),
    };

    return entity;
  }
